src/EventSourcingTests/Projections/Flattened/WriteTableWithGuidIdentifierProjection.cs

```diff
@@ -0,0 +1,152 @@
+using System;
+using Marten.Events.Projections.Flattened;
+
+namespace EventSourcingTests.Projections.Flattened;
+
+public class WriteTableWithGuidIdentifierProjection: FlatTableProjection
+{
+    public WriteTableWithGuidIdentifierProjection(): base("values", SchemaNameSource.DocumentSchema)
```

Comment:
**QUESTION:** I thought we would do something similar to what we have for regular projections, so allow someone to provide the view type and map by convention to the column names. We could even benefit from the code generation by doing that. We could enable people to configure mapping either through projection config and/or attributes and then just do upserts.

I think what's here is more flexible but also quite raw and different from what we have in other places. Yet, that doesn't mean that's bad. I think it'd be good to have something raw like that for someone who'd like to be more precise. So something like MongoDB API with their atomic operations. It might be even worth moving to regular JSON projections in the form of partial updates.

Reply:
Can you provide a concrete example? I'm not sure what about this solution isn't already pretty declarative. It's just "I want these members of these event types upserted into this row". By default, the column names are even derived from the member names. The point of this was just to write data to a straight SQL table. The examples I'm going to use it for are 1.) to help group a big cross-stream aggregation and 2.) to support metrics in a app where a flat table is advantageous to hit with SQL.

I'm definitely not going for a real ORM here.

And this does use code generation behind the scenes of course.